user/what-is-a-workspace.rst

```diff
@@ -0,0 +1,314 @@
+It builds and tests those packages in a **colcon workspace**, but what is a workspace?
+
+What's in a workspace?
+----------------------
+
+A workspace has these parts:
+
+* Software packages
+* Build artifacts
+* Logs
+* Install artifacts
+
+All these parts are put into different subdirectories of single directory, called the **workspace root**.
+Lets create a single directory for our workspace.
+
+.. code-block:: bash
+
+	mkdir ws
+
+Go into the root of our new workspace.
+
+.. code-block:: bash
+
+	cd ws
+
+
+Software packages
+*****************
+
+A workspace needs the source code of all the software to be built.
+Colcon will search all subdirectories of the workspace to look for packages, but an established convention is to put all the packages into a directory called ``src``.
+Let's create a directory for source code.
+
+.. code-block:: bash
+
+	mkdir src
+
+We'll need at least one software package in the workspace.
+Let's create a Python package.
+
+.. code-block:: bash
+
+	mkdir src/foo
+	touch src/foo/setup.py
+	touch src/foo/foo.py
+
+
+Put this content into ``src/foo/setup.py``:
+
+.. code-block:: python
+
+	from setuptools import setup
+
+	setup()
+
+Put this content into ``src/foo/setup.cfg``:
+
+.. code-block:: ini
+
+	[metadata]
+	name = foo
+	version = 0.1.0
+
+	[options]
+	py_modules =
+	  foo
+	zip_safe = true
+
+	[options.extras_require]
+	test =
+	  pytest
+
+	[tool:pytest]
+	junit_suite_name = foo
+
+Put this content into ``src/foo/foo.py``:
+
+.. code-block:: python
+
+	def foo_func():
+		print('Hello from foo.py')
+		return True
+
+
+Build artifacts
+***************
+
+The software build process often produces intermediate build artifacts.
+They are usually not used directly, but keeping them around makes subsequent builds faster.
+Colcon always directs packages to build out-of-source, meaning the build artifacts are put into a directory separate from the source code.
+Every package gets its own build directory, but all build directories are put into a single base directory.
+That single directory is called the ``build base``, and by default it's named  ``build`` at the root of the workspace.
+
+.. note::
+
+	You can change where build artifacts are put using the ``--build-base`` option to ``colcon build``.
+
+Lets build the software and see its build artifacts.
+
+.. code-block:: bash
+
+	# Make sure you run this command from the root of the workspace!
+	colcon build
+
+You'll see these new directories: ``build``, ``install``, and ``log``.
+
+::
+
+	ws
+	├── build
+	│	├── COLCON_IGNORE
+	│	└── foo/...
+	├── install/...
+	├── log/...
+	└── src
+	    └── foo
+	        ├── foo.py
+	        ├── setup.cfg
+	        └── setup.py
+
+Notice the ``build`` directory has a subdirectory ``foo`` and a file ``COLCON_IGNORE``.
+The ``foo`` subdirectory has all the build artifacts produced when building ``foo``.
+The ``COLCON_IGNORE`` file tells colcon there are no software packages in this directory.
+
+Logs
```

Comment:
So interestingly, this section is somewhat of a duplicate of https://github.com/colcon/colcon.readthedocs.org/blob/main/user/how-to.rst#log-files-of-past-invocations .  In point of fact, I think that the documentation from the `how-to.rst` file more appropriately belongs here, so my suggestion would be to delete it from `how-to.rst` and move it here.  As a side-note, it also describes what the `events.log` and `logger_all.log` files are for.